authnz/src/main/java/io/stargate/auth/AuthenticationPrincipal.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.auth;
+
+public class AuthenticationPrincipal {
+
+  private String token;
```

Comment:
Since we're keeping credentials here, this object looks more like a "Subject" to me, cf. `javax.security.auth.Subject`. How about renaming to `AuthenticationSubject` or perhaps just to `Subject` since clashes with JRE are not likely in this context, I guess.

Reply:
Also, `javax.security.auth.login.LoginContext` produces a `Subjet` on successful login.

Reply:
Yeah that makes sense. Will rename to AuthenticationSubject